### PR TITLE
fix: Ensure federation can handle bigger on chain deposits

### LINKED ISSUE #2258: Ensure federation can handle bigger on chain deposits.
and amounts in general.

In https://github.com/fedimint/fedimint/issues/1448 we've discovered that we might break when too much funds are deposited at once, as the current code can't handle too many notes due to artificial limits.

Write a test depositing 100BTC, some sends between users and fix any errors.

### PR DESCRIPTION
**Pull request by jkitman**
Fixes #2258

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /target/
 # in case it's a symlink
 /target
+fedimint-client/target/
```
Comment (dpc) [inline .gitignore:4]: This should no longer be needed. You might just have an old file from before https://github.com/fedimint/fedimint/pull/2286 . Or the new release didn't actually fix it. But they said they did.

File: fedimint-core/src/tiered_multi.rs [hunk 2]
```diff
@@ -124,9 +124,20 @@ impl<T> TieredMulti<T> {
...
+            .iter()
+            .filter_map(|(amt, notes)| {
+                if amt != except {
+                    Some(notes.len())
+                } else {
+                    None
+                }
+            })
+            .max()
+            .unwrap_or(0)
```
Comment (douglaz) [inline fedimint-core/src/tiered_multi.rs:140]: nit:
```suggestion
            .unwrap_or_default()
```

File: fedimint-core/src/tiered_multi.rs [hunk 3]
```diff
@@ -124,9 +124,20 @@ impl<T> TieredMulti<T> {
             .flat_map(|(amt, notes)| notes.into_iter().map(move |c| (amt, c)))
     }
 
-    /// Returns the length of the longest vector of all tiers
-    pub fn longest_tier_len(&self) -> usize {
-        self.0.values().map(|notes| notes.len()).max().unwrap_or(0)
+    /// Returns the length of the longest vector of all tiers, except the max
```
Comment (douglaz) [inline fedimint-core/src/tiered_multi.rs:127]:
```suggestion
    /// Returns the length of the longest vector of all tiers, except the `except`
```

File: .gitignore (hunk 1 above)
Comment (jkitman) [reply .gitignore:4]: Didn't seem to fix it for me :frowning_face:

Comment (jkitman) [reply .gitignore]: Or maybe it did?  I'll remove it from this PR.

**Review by elsirion — APPROVED** [review]

**Pull request merged**